library/client_events.js

```diff
@@ -116,6 +117,24 @@ module.exports = client => {
             client.createGuild(newMember.guild.id)
             client.log('Created new guild.')
           } else {
+            // if in any practice channel, member has rights to speak in practice room chat (can be muted)
+            let prChan = client.guilds.get(newMember.guild.id).channels.find(chan => chan.name === 'practice-room-chat')
+            if (prChan == null) {
+              client.log('Cannot find #practice-room-chat!')
+            } else if (restwo.permitted_channels.includes(newMember.voiceChannelID)) {
+              prChan.overwritePermissions(newMember.user, { SEND_MESSAGES: true })
+            } else {
+              let existingOverride = prChan.permissionOverwrites.get(newMember.user.id)
+              // existingOverride shouldn't be null unless someone manually deletes the override, but if for some reason it's gone, no big deal, just move on.
+              if (existingOverride != null) {
+                if (existingOverride.allowed.bitfield === Discord.Permissions.FLAGS.SEND_MESSAGES && existingOverride.denied.bitfield === 0) { // the only permission was allow SEND_MESSAGES
+                  existingOverride.delete()
```

Comment:
I'll assume that somehow magically it gets removed from `prChan.persmissionOverwrites`.

Reply:
Actually, it *doesn't*. All `delete()` does is send a REST message to delete the overwrite from the server. The local objects don't get updated at all, which is why we can't call `overwritePermissions` and *then* check if the arrays are empty - because the arrays never empty. The reason why this doesn't matter to us is because we never bother using any of these objects again.

Does this change anything?

(tbh this doesn't seem like the best design on the part of discord.js)